quadsim/robots/robot.py

```diff
@@ -1,44 +1,222 @@
-from typing import Any
-from typing import List
+"""Base class for all robots."""
+import numpy as np
+import pybullet_data
 
-# TODO: move Sensor and Motor to appropriate files
-
-
-class Sensor:
-    pass
-
-
-class Motor:
-    pass
+from quadsim.robots import motor_model
 
 
 class Robot:
     """Robot Base."""
 
-    def __init__(
-        self,
-        pybullet_client: Any,
-        motors: List[Motor],
-        sensors: List[Sensor],
-        on_rack: bool = False,
-    ) -> None:
+    def __init__(self, pybullet_client, config) -> None:
         """Constructs a base robot and resets it to the initial states.
 
         Args:
-
           pybullet_client: The instance of BulletClient to manage different
             simulations.
           motors: A list of motor models.
           sensors: A list of sensor models.
           on_rack: Whether the robot is hanging in mid-air or not. Used for debugging.
```

Comment:
Another thing we should decide is how we want to do docstrings. I think maybe they're most important for just the big classes like the `Robot` base class. What do you think?

Reply:
My general idea is:
1) Code should be self-documenting for the most parts (i.e. one can understand most of the code by reading only the code)
2) The documentation should focus on high-level ideas that are not obvious during first read. (e.g. the bullet action_repeat/timestep, the urdf_ids and how they work).

What do you think?

Reply:
Perfect. I'm on the exact same page =].

Reply:
btw, I think including type hints in all the major classes will go a long way to being 'self-documenting' 😄

Reply:
Agree. Did I miss type hints at some major places? (Sorry it's my first time using python typing).

Reply:
other than the headers taking configs (which will change), it's looking good so far =]!